core/src/apps/nem/mosaic/layout.py

```diff
@@ -34,57 +31,50 @@ async def ask_mosaic_creation(
 async def ask_supply_change(
     ctx: Context, common: NEMTransactionCommon, change: NEMMosaicSupplyChange
 ) -> None:
-    await require_confirm_content(ctx, "Supply change", _supply_message(change))
+    from trezor.enums import NEMSupplyChangeType
+    from ..layout import require_confirm_text
+
+    supply_message = [
+        ("Modify supply for", change.mosaic),
+        ("under namespace", change.namespace),
+    ]
+    await require_confirm_content(ctx, "Supply change", supply_message)
+    common_msg = " supply by " + str(change.delta) + " whole units?"
     if change.type == NEMSupplyChangeType.SupplyChange_Decrease:
-        msg = "Decrease supply by " + str(change.delta) + " whole units?"
+        msg = "Decrease" + common_msg
     elif change.type == NEMSupplyChangeType.SupplyChange_Increase:
-        msg = "Increase supply by " + str(change.delta) + " whole units?"
+        msg = "Increase" + common_msg
     else:
```

Comment:
Nice, this saved `30` bytes. I was afraid about the `f-strings` because in the past I noticed they are more costly than string concatenation by `+`, but that is not the case here.

Reply:
Taking that back, it indeed saved 30 bytes in `ask_supply_change()` function, but the overall size of `str1` increased by the same exact 30 bytes. The same case as with the `ripple`'s example above. So it all might in the end compile to the same bytecode, even though the sizes are distributed elsewhere

Reply:
Right, so in the end it is beneficial, but most of the time not visible in overall alligned flash size